### DISCUSSION TITLE
Recurring notification for the same resources

### DISCUSSION THREAD

**Discussion by ViBiOh** [fluxcd/flux2#1049] [Q&A] [answered]
Hi 👋 

We have started our migration from `flux` to `flux2` and we encounter some behaviors with notification that we didn't expected. Each time `flux2` reconcile, the Kustomization Controller emits a notification for configuring resources that are already created.

For example, we have this manifest in our git repository.

```yaml
apiVersion: scheduling.k8s.io/v1
kind: PriorityClass
metadata:
  name: high-priority-db
value: 100
globalDefault: false
description: "This priority class is reserved for DB. It gives the priority over other workload to be scheduled."
```

If I retrieve the value from the cluster, I got this

```
> kubectl get priorityclasses.scheduling.k8s.io high-priority-db -o yaml
apiVersion: scheduling.k8s.io/v1
description: This priority class is reserved for DB. It gives the priority over other
  workload to be scheduled.
kind: PriorityClass
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"scheduling.k8s.io/v1","description":"This priority class is reserved for DB. It gives the priority over other workload to be scheduled.","globalDefault":false,"kind":"PriorityClass","metadata":{"annotations":{},"labels":{"kustomize.toolkit.fluxcd.io/checksum":"960782492a141cafaed089f15d91f0586374536a","kustomize.toolkit.fluxcd.io/name":"kube-manifests","kustomize.toolkit.fluxcd.io/namespace":"flux-system"},"name":"high-priority-db"},"value":100}
  creationTimestamp: "2020-06-24T13:06:57Z"
  generation: 1
  labels:
    kustomize.toolkit.fluxcd.io/checksum: 960782492a141cafaed089f15d91f0586374536a
    kustomize.toolkit.fluxcd.io/name: kube-manifests
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: high-priority-db
  resourceVersion: "549090289"
  selfLink: /apis/scheduling.k8s.io/v1/priorityclasses/high-priority-db
  uid: 61bf7337-802a-403f-8ff6-03a3dc6c535c
preemptionPolicy: PreemptLowerPriority
value: 100
```

Apart from Flux / Kubernetes sugar on labels and annotation, there is nothing new, except `preemptionPolicy` maybe.

Our configuration for alert is the following

```
apiVersion: notification.toolkit.fluxcd.io/v1beta1
kind: Alert
metadata:
  name: slack
  namespace: flux-system
spec:
  providerRef:
    name: slack
  eventSeverity: info
  eventSources:
    - kind: Kustomization
      namespace: flux-system
      name: "*"
```

And each time kustomize is running, I got a slack message (along with all the CRDs and some extra resources like this).

![Screen Shot 2021-03-05 at 2 53 03 PM](https://user-images.githubusercontent.com/2349470/110124558-872e0680-7dc2-11eb-8a5b-3ac8afdebe72.png)

How can I manage to have only real creation/update/deletion on notification instead of this notification ? Coming from `flux` and `fluxcloud` we didn't encounter this kind of behavior, each notification was real change.

**Comment by stefanprodan** [comment, marked as answer]
Please see https://github.com/fluxcd/flux2/issues/626

**Reply by ViBiOh** [reply]
Oh! I've searched but maybe with incorrect keywords, thanks I'll try 👍

**Reply by ViBiOh** [reply]
If I understand it well, adding the annotation `kustomize.toolkit.fluxcd.io/prune: disabled` will disable pruning.

Pruning is the "Garbage collector" (both terms are referenced in the documentation), the fact that if I delete it from git, it will be deleted from the cluster.

So, when disabling pruning (for notification reason), I can end up with objects never being deleted? Am I wrong?

**Reply by stefanprodan** [reply]
That annotation would not help, you need to disable it globally with `prune: false`.

**Reply by ViBiOh** [reply]
But in the end it's the same result, my resources won't be deleted if `prune: false`?

There is no way for getting notification of "resources changes" but conserving the GitOps philosophy ("if it's in git, it's applied, if no more, deleted")?
